### PR TITLE
FormPart: Add wrapper for content type

### LINKED ISSUE #7180: FormPart - Add wrapper for content type
`FormPart` is rendered using `Form.Wrapper.cshtml` for all content type containing `FormPart` 

Add wrapper in driver to allow customize form per content type.  

i.e. The custom Form content type  `SignupLead` containing `FormPart` can have shape template `Form-SignupLead.Wrapper.cshtml`.  where developer can provide custom form processing without affecting default implementation for `Form` content type or any other content type containing `FormPart`.

https://github.com/OrchardCMS/OrchardCore/issues/6326#issuecomment-670632804

### PR DESCRIPTION
**Pull request by ns8482e**
Fixes #7180  Add wrapper for content type

**(Updated)** For example, following wrapper template `Form-SignupLead.Wrapper.cshtml` for content type `SignupLead` enables form submitted to workflow via ajax - fetch api.

```cshtml
@using OrchardCore.Mvc.Utilities;


@{
    dynamic formPart = Model.ContentItem.Content.FormPart; 
}

<form action="@formPart.Action" 
      method="@formPart.Method"  enctype="@formPart.encType" 
      class="form-content form-@((Model.ContentItem.ContentType as string).HtmlClassify())"
      onsubmit="ajaxSubmit(event)">
    @if(formPart.EnableAntiForgeryToken.Value as bool? ?? false)
    {
        @Html.AntiForgeryToken()
    }

    @await DisplayAsync(Model.Metadata.ChildContent)

    <input type="hidden" name="customHidden1" value="value1" />
    <input type="hidden" name="customHidden1" value="value1" />

    <script at="Foot">
        function ajaxSubmit(e){

                e.preventDefault();

                const formJson = [].reduce.call(e.target.elements, function(data, element) {
                    data[element.name] = element.value;
                    return data;
                }, {});
                
                var url = e.target.getAttribute('action');  
                fetch(url, {
                    method: e.target.getAttribute('method'),
                    headers: {
                        'Content-Type': 'application/json'  
                    },
                    body: JSON.stringify(formJson) 
                }).then( function(resp){
                    // process response
                });
                
            }
    </script>
</form>
```

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs [hunk 1]
```diff
@@ -13,10 +13,14 @@ public class FormContentDisplayDriver : ContentDisplayDriver
...
         {
             var formItemShape = context.Shape;
-            // If the content item contains FormPart add Form Wrapper.
+            // If the content item contains FormPart add Form Wrapper only in Display type Detail
             var formPart = model.As<FormPart>();
-            if (formPart != null)
+            if (formPart != null && context.DisplayType == "Detail")
             {
+                // Add wrapper for content type
+                formItemShape.Metadata.Wrappers.Add($"Form_Wrapper__{model.ContentType}");
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: Looks like this gets hit twice, and is generating two `<form>` tags as a result

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: All wrappers are rendered. here `Form_Wrapper` will generate form Tag and `Form_Wrapper__[content type]` will generate inner wrapper that allows to provide additional customization within that form. No need to define form tag again

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: Try with my example code above

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: ah, but have you tried it without creating the second wrapper template?

Try with the coming soon theme. two form tags...

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: @deanmarcussen  It's not related to this PR, May be you have left over old code for ComingSoon. You  have already fixed that earlier in PR #6346

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: Flow module fresh install.

Before 1 form tag:

Screenshot 2020-10-04 at 09 42 06

After 2 form tags:

Screenshot 2020-10-04 at 09 43 07

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: Wow! It seems wrapper fall back to parent shape

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:21]: fixed

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:20]: Now you got it ;)

Looks good. Why did you remove

`id="form_@Model.ContentItem.ContentItemId"`

I thought that looked useful?

Comment (ns8482e) [reply src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormContentDisplayDriver.cs:20]: I removed id as I feel that its better to use `FormElementPart` but It needs changes in migration so will create separate PR

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**